minitwit/src/main/java/WebApplication.java

```diff
@@ -665,16 +667,67 @@ else if (!BCrypt.checkpw(enteredPW, rs.getString("pw_hash"))) {
         updateLatest(request);
 
         var messages = getMessages();
-        var filteredMessages = messages.stream().map((message) -> {
-            return Map.ofEntries(
-                    Map.entry("content", message.get("text")),
-                    Map.entry("pub_date", message.get("pub_date")),
-                    Map.entry("user", message.get("username"))
-            );
-        });
+        var filteredMessages = messages.stream().map((message) -> Map.ofEntries(
+               Map.entry("content", message.get("text")),
+               Map.entry("pub_date", message.get("pub_date")),
+               Map.entry("user", message.get("username"))
+       ));
         return filteredMessages.toList();
     };
 
+    //  --------------------------------------
+    //              Simulator API
+    //  ---------------------------------------
+
+    public static Route serveSimMsgsUsr = (Request request, Response response) -> {
+
+        updateLatest(request);
+
+        SQLite db = new SQLite();
+        var connection = db.getConnection();
+        var userID = getUserID(db, request.params(":username"));
+
+        int noMsgs =  Integer.parseInt(request.queryParamOrDefault("no", "100"));
+
+        if (request.requestMethod().equals("GET")){
+            if (userID == 0) {
+                halt(404, "404: Could not find user!"); //How do we treat 404?
```

Comment:
I think we can just return
response.status(404);

Or is there another status that reflects better that the user does not exist

Reply:
Hmm maybe. In the API specification it uses flask's abort. Not entirely sure that halt functions the same though, but it seemed the most relevant to me .